fitbenchmarking/fitbenchmarking
# Refactor jacobian and hessian information

#### Description of Work

Fixes #901 


#### Testing Instructions

1. Check that jacobians/hessians are still only used for minimizers that accept them
2.
3.

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
